compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -1063,3 +1065,32 @@ impl<'tcx> Stable<'tcx> for rustc_middle::ty::BoundTy {
         BoundTy { var: self.var.as_usize(), kind: self.kind.stable(tables) }
     }
 }
+
+impl<'tcx> Stable<'tcx> for mir::interpret::Allocation {
+    type T = stable_mir::ty::Allocation;
+
+    fn stable(&self, tables: &mut Tables<'tcx>) -> Self::T {
+        let start = rustc_target::abi::Size::ZERO;
+        let size = self.size();
+        let range = mir::interpret::AllocRange { start, size };
+        let mut blocks = Vec::new();
+        for i in self.init_mask().get_blocks() {
+            blocks.push(*i);
+        }
+        stable_mir::ty::Allocation {
+            bytes: self.get_bytes_unchecked(range).to_vec(),
+            provenance: {
+                let mut ptrs = Vec::new();
+                for (size, prov) in self.provenance().ptrs().iter() {
+                    ptrs.push((size.bytes_usize(), opaque(&prov.0.get())));
```

Comment:
```suggestion
                    ptrs.push((size.bytes_usize(), opaque(prov)));
```

when we render it opaquely, just render the full thing instead of its internal information